docs/pci/contribution-guidelines.md

```diff
@@ -0,0 +1,219 @@
+# PCIe Support in Firecracker Community Roadmap
+
+This document describes the high-level changes required to support PCIe and device passthrough in Firecracker
+and the main responsibilities of the maintainers and the community to achieve the success of the initiative.
+This document will be discussed during the November 6, 2024 meeting.
+I will upload this document as a PR to the [poc/pcie](https://github.com/firecracker-microvm/firecracker/tree/poc/pcie)
+branch so that everybody will have the opportunity to leave comments along the way.
+
+## Motivation
+
+Firecracker currently supports only MMIO devices. 
+By adding support for PCIe we would get the following benefits:
+
+* Increase max attached device count - up to 31 devices on a single PCI bus, with up to 256 buses, 
+  if we add support for multiple buses.
+* Ability to assign multiple interrupts per device (MSI-X) - opens the door for I/O scalability / MQ devices
+    * MSI-X interrupts also improve virtio-pci performance over virtio-mmio which uses legacy IRQ
+* device hot-plugging through PCIe Hotplug
+* pass-through of physical devices, like GPUs or EBS volumes, through VFIO.
+
+## Challenges
+
+Supporting PCIe in Firecracker and, in particular, device pass-through, introduces new challenges. Namely:
+
+* **overheads:** supporting the full PCI specification might negatively impact the boot time and 
+  memory overheads of Firecracker VMs. 
+    * We can mitigate this by allowing for completely disabling PCIe support via VM configuration 
```

Comment:
Could this be done on-the fly on a per-vm basis, our would it require using a specific build of firecracker?

Reply:
we were thinking a built time flag. In this way it is a conscious decision to enable PCIe, once is enabled Firecracker will make use of it.